### ISSUE TITLE
Add analyzer for gigasecond

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/javascript-analyzer#32]
This issue is for discussion and assignment for the [`gigasecond`][exercise] core exercise in the `javascript` track. 

🔗 [implementation][exercise] | [mentor-notes][mentor-notes] | [problem-specification][problem-specification]

----

[mentor-notes]: https://github.com/exercism/website-copy/tree/master/tracks/javascript/exercises/gigasecond/mentoring.md
[exercise]: https://github.com/exercism/javascript/tree/master/exercises/gigasecond
[problem-specification]: https://github.com/exercism/problem-specifications/tree/master/exercises/gigasecond

This exercise introduces [`Date`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) and `Math.pow` (`**` operator).

```javascript
const GIGASECOND_IN_MS = (10 ** 9) * 1000

export function gigasecond(date) {
  return new Date(date.getTime() + GIGASECOND_IN_MS)
}
```

## SHOULD comment and disapprove

Cases we _SHOULD_ comment on with this analyzer:
- Using `Date#valueOf`
- Using `Date.parse(date)`
- Using mutation (`setSeconds`)
- Using `**` or `Math.pow`

### COULD comment (tips)

Cases we _COULD_ comment on with this analyzer:
- Doing an `export` with a specifier (at the bottom of the file) can be done `export const` inline

**Comment by SleeplessByte** [comment]
## Raw output

```json
{
  "invalid": 0,
  "valid": 496,
  "total": 496,
  "unique": 421
}
```

## Parsing statistics

This is the number of unique Abstract Syntax Trees after stripping commentary,
location data (whitespace) and other tokens.

|   total |  unique |   valid | invalid |
|--------:|--------:|--------:|--------:|
|     496 |     421 |     496 |       0 |

**Comment by SleeplessByte** [comment]
Initial `approve_as_optimal` only analyzer

|               Status | Count | Comments | Unique |    Avg | Median |   Total |
| --------------------:| -----:| --------:| ------:| ------:|-------:|--------:|
|    Approve (optimal) |     5 |        0 |      0 |  238ms |  246ms |      0s |
|    Approve (comment) |     0 |        0 |      0 |    0ms |    0ms |      0s |
| Disapprove (comment) |    34 |        4 |      4 |  194ms |  232ms |      0s |
|      Refer to mentor |   456 |        0 |      0 |  180ms |  221ms |      8s |
|                Total |   495 |        4 |      4 |  181ms |  225ms |      8s |

**Comment by SleeplessByte** [comment]
Closed by #38 


## Statistics

|               Status | Count | Comments | Unique |    Avg | Median |   Total |
| --------------------:| -----:| --------:| ------:| ------:|-------:|--------:|
|              Approve |    86 |       28 |      4 |  785ms |  374ms |      6s |
|           Disapprove |   234 |       44 |     12 |  749ms |  480ms |     17s |
|      Refer to mentor |   176 |       37 |      4 |  729ms |  488ms |     12s |
|                Total |   496 |       91 |     14 |  748ms |  465ms |     37s |
